### PR TITLE
Enable template localization on dotnet CLI

### PR DESCRIPTION
**Pull request by bekir-ozturk**
## Background:
 Template engine (`dotnet new`) added support for template localizations in .NET 6.0. This new system replaces the old way of localizing templates that only worked on Visual Studio and works on both .NET CLI as well as VS.

This PR introduces changes to switch to the new template localization system.

## Summary of the changes
- Adds `<UsingToolTemplateLocalizer>true</>` to projects containing templates.
- Adds necessary `id` fields into `template.json` files to allow localizing certain fields (specifically, post actions).
- Generates localization files to be translated by loc team

## What to expect after merging
Every time there is a change to one of the templates:
- The dev making the change should build (at the very least) the modified template project. This will update the loc files on the local working copy,
- Push the loc files together with the template modifications. Review & merge.
- OneLocBuild integration will automatically pick up the changes and will send them for translation.
- You will receive a PR containing the translated template loc files when they are ready. Review & merge.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6319)

### CODE REVIEW

**Review by vlada-shubina — DISMISSED** [review]

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.es.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
-﻿{
+{
   "author": "Microsoft",
   "name": "Aplicación de Windows Forms",
-  "description": "Plantilla de proyecto para crear una aplicación de Windows Forms (WinForms) de .NET.",
+  "description": "Plantilla de proyecto para crear una aplicación de Windows\u00A0Forms (WinForms) de .NET.",
```
Comment (RussKie) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.es.json:4]: Why was this changed?

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/Microsoft.Dotnet.Winforms.ProjectTemplates.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
     <TargetFramework>$(TargetFrameworkName)$(TargetFrameworkVersion)</TargetFramework>
     <EnableDefaultItems>false</EnableDefaultItems>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
-    
+    <UsingToolTemplateLocalizer>true</UsingToolTemplateLocalizer>
```
Comment (RussKie) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/Microsoft.Dotnet.Winforms.ProjectTemplates.csproj:9]: Trailing spaces

**Review by RussKie — COMMENTED** [review]
There appears to be a lot of "noisy" changes, that don't carry any value.

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.es.json (hunk 1 above)
Comment (bekir-ozturk) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.es.json:4]: In some cases, loc team used `hard space`s that won't break into a new line instead of a regular whitespace character. Our automated tool is configured to encode such symbols into `\u####` form.
Hopefully, the tooling they are using will be able to deal with this change. But if starts going back and forth, we will update the tool to preserve the current encoding.

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/Microsoft.Dotnet.Winforms.ProjectTemplates.csproj (hunk 2 above)
Comment (bekir-ozturk) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/Microsoft.Dotnet.Winforms.ProjectTemplates.csproj]: Removed.

**Review by RussKie — APPROVED** [review]

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.es.json (hunk 1 above)
Comment (dreddy-work) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsApplication-CSharp/.template.config/localize/templatestrings.es.json:4]: @bekir-ozturk , this change is causing the problem. Loc pipeline converted this back to its original state removing the `\u####`. But local build adds it back and thus every time you sync with main and try to build repo locally, you will be getting this unwanted change as modified one. Is this change communicated to loc pipeline?
